src/geometry.F90

```diff
@@ -76,20 +76,18 @@ subroutine check_cell_overlap(p)
     type(Particle), intent(inout) :: p
 
     integer :: i                       ! cell loop index on a level
+    integer :: j
+    integer :: n_coord
```

Comment:
Oops, I see that you already have a ``n_coord`` here, so perhaps it won't be as easy to intuitively name "j". If there isn't a good name, at least add a comment explaining what this var is for here